### PR TITLE
Update to Go 1.19 on release/0.4.x

### PR DESCRIPTION
**Pull request by pglass**
## Changes proposed in this PR:
This updates to Go 1.19 on **release/0.4.x**. This requires fixes a couple of things to satisfy golangci-lint:

- Re-run go fmt, which is a noisy change because Go 1.19 formats comments.
- Stop using ioutil which is deprecated
- Also, update golangci-lint

## Checklist:
- [ ] Tests added
- [ ] CHANGELOG entry added

    [HashiCorp engineers only. Community PRs should not add a changelog entry.]::
    [Changelog entries should use present tense, e.g. "Add support for..."]::

### CODE REVIEW

**Review by cthain — APPROVED** [review]

File: .github/workflows/bin-ci.yml [hunk 1]
```diff
@@ -39,14 +55,14 @@ jobs:
         - 1.11.11+ent
     env:
       TEST_RESULTS_DIR: /tmp/test-results
-      GOTESTSUM_VERSION: 1.8.0
+      GOTESTSUM_VERSION: 1.8.2
```
Comment (wilkermichael) [inline .github/workflows/bin-ci.yml:58]: I think you need to add a dependency on test for `get-go-version` similar to what you did for lint

**Review by wilkermichael — APPROVED** [review]
Looks good to me, just one fix

File: .github/workflows/bin-ci.yml (hunk 1 above)
Comment (pglass) [reply .github/workflows/bin-ci.yml:60]: Ah, yes. It actually seemed to default to go 1.18 without this. Thank you!

Comment (pglass) [reply .github/workflows/bin-ci.yml:60]: I already merged the PRs for the other release branches, so I picked and push this fix to those branches:

* release/0.5.x - https://github.com/hashicorp/consul-ecs/commit/601380b2d325a826e2fdd0d9f154545685b84e9a
* main - https://github.com/hashicorp/consul-ecs/commit/473cdaf50b7ee016cd21b01c4cfb18572611a4ce

**Pull request merged**